integration-tests/bats/json-diff.bats

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+
+    dolt sql <<SQL
+CREATE TABLE test (
+  pk BIGINT NOT NULL COMMENT 'tag:0',
+  c1 BIGINT COMMENT 'tag:1',
+  c2 BIGINT COMMENT 'tag:2',
+  c3 BIGINT COMMENT 'tag:3',
+  c4 BIGINT COMMENT 'tag:4',
+  c5 BIGINT COMMENT 'tag:5',
+  PRIMARY KEY (pk)
+);
+SQL
+}
+
+teardown() {
+    assert_feature_version
+    teardown_common
+}
+
+@test "json-diff: new row" {
+    dolt add .
+    dolt commit -m table
+    dolt sql -q 'insert into test values (0,0,0,0,0,0)'
+    dolt add .
+    dolt commit -m row
+
+    dolt diff -r json
+    run dolt diff -r json
+    [ "$status" -eq 0 ]
+    [ "$output" = "" ]
+
+    run dolt diff -r json head
+    [ "$status" -eq 0 ]
+    [ "$output" = "" ]
+
+    dolt diff -r json head^
+    run dolt diff -r json head^
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ '{"tables":[{"name":"test","schema_diff":[],"data_diff":[{"from_row":{},"to_row":{"c1":0,"c2":0,"c3":0,"c4":0,"c5":0,"pk":0}}]}]}' ]] || false
+
+    dolt diff -r json head^ head
+    run dolt diff -r json head^ head
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ '{"tables":[{"name":"test","schema_diff":[],"data_diff":[{"from_row":{},"to_row":{"c1":0,"c2":0,"c3":0,"c4":0,"c5":0,"pk":0}}]}]}' ]] || false
+
+    dolt diff -r json head head^
+    run dolt diff -r json head head^
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ '{"tables":[{"name":"test","schema_diff":[],"data_diff":[{"from_row":{"c1":0,"c2":0,"c3":0,"c4":0,"c5":0,"pk":0},"to_row":{}}]}]}' ]] || false
+}
+
+@test "json-diff: data and schema changes" {
```

Comment:
I found an issue with schema+data changes while testing out a build...

I started with your BATS test table, then ran:
* ALTER TABLE `test` RENAME COLUMN `c2` TO `c2new`;
* update test set c2new = 1;

`dolt diff` shows:
```
+---+----+----+------+----+----+----+-------+
|   | pk | c1 | c2   | c3 | c4 | c5 | c2new |
+---+----+----+------+----+----+----+-------+
| < | 0  | 0  | 0    | 0  | 0  | 0  | NULL  |
| > | 0  | 0  | NULL | 0  | 0  | 0  | 1     |
+---+----+----+------+----+----+----+-------+
```

But `dolt diff -r json` shows:
```
 {
          "from_row": {
            "c1": 0,
            "c2new": 0,
            "c3": 0,
            "c4": 0,
            "c5": 0,
            "pk": 0
          },
          "to_row": {
            "c1": 0,
            "c3": 0,
            "c4": 0,
            "c5": 0,
            "pk": 0
          }
}
```

I would expect the c2new column to be listed in the to_row with the value `1`.

Reply:
Good catch, fixed and tests added.